### PR TITLE
feat(docs): Add initial `/package` & `/package/[name]` routes

### PR DESCRIPTION
**Pull request by xeho91**
While working on #56 I wanted to create a route `/package/[name]/error/[name]`.

It couldn't be done without creating the parent routes first.

For starters, I think it's also best practice to use as much metadata as possible.
So we can directly focus on the content inside the `/packages` directory.

In the current state, what's used for each package route is:

- `package.json`
- `README.md` content

Oh and also, I've removed `/examples` in favour of `package/*`

### CODE REVIEW

**Comment by xeho91** [comment]
@ciscorn  Is there any way I could see the logs from Cloudflare Pages building?

I suspect the errors might be related to usage of `node:*` modules.

**Comment by ciscorn** [comment]
@xeho91 Unfortunately, there isn’t an easy way for you to view the logs from Cloudflare Pages builds, as Cloudflare Pages doesn’t offer granular permissions for individual deployments. Essentially, it’s an all-or-nothing access to my account.

Here are the logs:

```
17:01:13.316 | file:///opt/buildhome/repo/apps/docs/.svelte-kit/output/server/entries/pages/package/_name_/_page.server.ts.js:2
-- | --
17:01:13.316 | import path$1, { join, dirname, win32, resolve as resolve$2 } from "pathe";
17:01:13.317 | ^^^^^
17:01:13.317 | SyntaxError: The requested module 'pathe' does not provide an export named 'win32'
17:01:13.317 | at ModuleJob._instantiate (node:internal/modules/esm/module_job:171:21)
17:01:13.317 | at async ModuleJob.run (node:internal/modules/esm/module_job:254:5)
17:01:13.317 | at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:482:26)
17:01:13.317 | at async Promise.all (index 8)
17:01:13.317 | at async analyse (file:///opt/buildhome/repo/node_modules/.pnpm/@sveltejs+kit@2.14.0_@sveltejs+vite-plugin-svelte@5.0.3_svelte@5.14.4_vite@6.0.5_@types+node@_bctgs6rfkwdn6dzcwakpnp7huq/node_modules/@sveltejs/kit/src/core/postbuild/analyse.js:74:16)
17:01:13.317 | at async MessagePort.<anonymous> (file:///opt/buildhome/repo/node_modules/.pnpm/@sveltejs+kit@2.14.0_@sveltejs+vite-plugin-svelte@5.0.3_svelte@5.14.4_vite@6.0.5_@types+node@_bctgs6rfkwdn6dzcwakpnp7huq/node_modules/@sveltejs/kit/src/utils/fork.js:23:16)
17:01:13.317 |  
17:01:13.317 | Node.js v22.8.0
17:01:13.392 | ELIFECYCLE  Command failed with exit code 1.
17:01:13.424 | Failed: Error while executing user command. Exited with error code: 1
17:01:13.435 | Failed: build command exited with code: 1
17:01:14.243 | Failed: error occurred while running build command
```

File: apps/docs/src/routes/package/+layout.server.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import normalize from "normalize-package-data";
+
+const __filename = url.fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+export async function load() {
+	const root_dir_url = url.pathToFileURL(
+		path.resolve(__dirname, "..", "..", "..", "..", "..", "packages"),
+	);
+	const packages = new Map(
```
Comment (ciscorn) [inline apps/docs/src/routes/package/+layout.server.ts:14]: Hmm, runtime (not build time) file system access like this won’t work on at least Cloudflare Pages/Workers and is questionable even in Node environments like Vercel.

Comment (xeho91) [reply apps/docs/src/routes/package/+layout.server.ts:14]: Yeah, I need to rethink this part with file system access. I'll put it to a draft for now.

Comment (xeho91) [reply apps/docs/src/routes/package/+layout.server.ts]: I've changed to using Vite feature [`import.meta.glob()`](https://vite.dev/guide/features.html#glob-import). This should do the trick, I hope?

**Comment by xeho91** [comment]
@ciscorn Would it be okay to ask you to take over this PR?
Given the limitation of not being able to see the logs from Cloudflare Pages, I'm unable to figure out what's going wrong there.

I suspect currently it might be related to monorepo & Vite setup - Cloudflare Pages cannot see or blocks the access to `/packages/` directory.

**Comment by ciscorn** [comment]
@xeho91 Sure, no problem 🫡
I’ll take a look at it tonight!

**Comment by ciscorn** [comment]
I can reproduce the build error on my Mac by running `pnpm run build` in the `apps/docs`:

```
/apps/docs/.svelte-kit/output/server/entries/pages/package/_name_/_page.server.ts.js:7
import path$1, { join, dirname, win32, resolve as resolve$2 } from "pathe";
                                ^^^^^
SyntaxError: The requested module 'pathe' does not provide an export named 'win32'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:180:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:263:5)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:547:26)
    at async Promise.all (index 8)
```

**Comment by xeho91** [comment]
I admit I haven't tested with `pnpm build` script. My bad. Still, error message is puzzling me. Could it be really issue with `pathe`package?

I can investigate it soon, I'm on the way to my hometown and in around 1h I'll land in the coffee shop, so I can get my hands on my laptop finally.

File: apps/docs/src/routes/package/[name]/+page.server.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { error } from "@sveltejs/kit";
+import { mdToSvelte } from "@sveltepress/vite";
```
Comment (ciscorn) [inline apps/docs/src/routes/package/[name]/+page.server.ts:2]: This seems to be causing the error. It appears to be designed for Vite (to run during the build process) and is likely not intended to be included in the runtime build.

Comment (xeho91) [reply apps/docs/src/routes/package/[name]/+page.server.ts:2]: Oh, sweet as... thanks for the lead!

**Comment by xeho91** [comment]
Blocked by: https://github.com/unjs/pathe/pull/193

File: apps/docs/src/routes/package/[name]/+page.server.ts (hunk 2 above)
Comment (xeho91) [reply apps/docs/src/routes/package/[name]/+page.server.ts]: I took the opportunity to attempt to solve this case. The issue is that `win32` export doesn't exist in `pathe`. So, I created a PR to provide this export. This solved the issue with `pnpm run build` locally.

**To sum up:**

I know for sure that we need `pathe` to be used in the browser.

Do I understand correctly that for Cloudflare Pages we also need to use `pathe`⁣ — even inside the Vite build process?

Comment (xeho91) [reply apps/docs/src/routes/package/[name]/+page.server.ts]: > Do I understand correctly that for Cloudflare Pages we also need to use pathe⁣ — even inside the Vite build process?

I tested—it worked! The answer is **no**.

**Comment by xeho91** [comment]
@ciscorn There's one issue left, which I suspect might be on the Sveltepress side. I've [reported](https://discord.com/channels/994160573333913691/1322860018780868680) it on their Discord. But I'll leave you to decide if this should block us from moving further.

**Comment by ciscorn** [comment]
@xeho91 It seems like the styles might be broken 🤔 
image

**Comment by xeho91** [comment]
Yeah, that's what I meant in the previous comment. I realised now I wasn't descriptive enough.

My initial hopes was to help support svelte ecosystem and use what exist already. I fear the timing might be inconvenient for us _(holiday season and the project isn'tthat active)_, and I probably wouldn't be that much blocked with using existing UI libraries. Together with `shiki`, `remark`, `rehype`, etc.

tl;dr; I don't feel any productivity boost from using Sveltepress.

P.S. Do you mind hop up for a chat on Discord? I'd like to settle on decision regarding the UI components library.

**Comment by ciscorn** [comment]
I’ve been thinking the same thing for a while.

I’m out right now, but I’ll check Discord later. Sorry for the wait.

Here’s an example of my setup for reference: https://svelte-maplibre-gl.mierune.dev/examples/side-by-side
(SvelteKit + mdsvex + Shiki + shadcn-svelte)

**Comment by xeho91** [comment]
I'm sold on this setup.

I wanted to ask you which one of UI libraries you felt the most comfortable with. But I think I have the answer. I have a faith that based on your setup this will work better without too much digging into internals, like with Sveltepress.

I'm okay to **ditch** Sveltepress in another PR. I don't think we're losing too much, as most of the code is basically using SvelteKit features.
I can work on this another PR.

P.S. I'm in awe of how much effort was put into this project, looks slick, and those examples are amazing!!! You and your team definitely know how to show a value!

**Review by ciscorn — APPROVED** [review]

**Comment by ciscorn** [comment]
I’m glad to see how both projects are influencing each other.

If you’re okay with it, how about starting by copying this setup? I probably won’t have enough time again today until late evening, so it’s totally fine if you set it up yourself. (Our docs is still WIP, so there’s much room for improvement). Feel free to copy the styles as well.

Here’s a short explanation for the choice of components:

- SSR/SSG Framework → SvelteKit is the obvious choice.
- Markdown integration → MDsveX is specifically designed for embedding Svelte components in Markdown, which is ideal for documentation.
- Syntax highlighter → Shiki is modern and comes with neat default themes.
- UI components → The shadcn-svelte family (Bits UI and Melt UI) is well-maintained and already supports Svelte 5.

**Comment by ciscorn** [comment]
I think using the README as the package description in the docs is a great idea!

**Comment by xeho91** [comment]
> I’m glad to see how both projects are influencing each other.
> 
> If you’re okay with it, how about starting by copying this setup? I probably won’t have enough time again today until late evening, so it’s totally fine if you set it up yourself. (Our docs is still WIP, so there’s much room for improvement). Feel free to copy the styles as well.
> 
> Here’s a short explanation for the choice of components:
> 
> - SSR/SSG Framework → SvelteKit is the obvious choice.
> - Markdown integration → MDsveX is specifically designed for embedding Svelte components in Markdown, which is ideal for documentation.
> - Syntax highlighter → Shiki is modern and comes with neat default themes.
> - UI components → The shadcn-svelte family (Bits UI and Melt UI) is well-maintained and already supports Svelte 5. 

Please allow me to work on this PR. I'm about to finish my errands, then I can start working on it.
I'll be studying and copying the bits from svelte maplibre project. This will also allow me to have a chance to understand how it works, and get used to this setup faster ☺️
